tests/unit/LimitedRestorationPanel.spec.ts

```diff
@@ -1,102 +1,133 @@
-localVue.use(VueRouter)
+// suppress the "[Vuetify] Unable to locate target [data-app]" warning
+document.body.setAttribute('data-app', 'true')
+
+const vuetify = createVuetify()
 
 /**
  * Creates and mounts a blank, un-populated component
  */
 function createDefaultComponent (
   months = 0,
   maxNumberOfMonths = 24
-): Wrapper<LimitedRestorationPanel> {
+): VueWrapper {
   return mount(LimitedRestorationPanel, {
-    propsData: {
+    props: {
       months,
       maxNumberOfMonths
     },
     vuetify,
-    localVue
+    VueRouter
   })
 }
 
-describe('Initialize RelationshipsPanel component', () => {
-  it('loads the component', () => {
-    const wrapper: Wrapper<LimitedRestorationPanel> = createDefaultComponent()
+describe('Initialize LimitedRelationshipsPanel component', () => {
+  let wrapper: any
+
+  it('loads the component with default values', () => {
+    wrapper = createDefaultComponent()
 
     expect(wrapper.findComponent(LimitedRestorationPanel).exists()).toBe(true)
+    expect(wrapper.vm.$data.radioValue).toEqual('customMonths')
+    expect(wrapper.vm.$data.inputValue).toEqual('0')
 
-    wrapper.destroy()
+    wrapper.unmount()
   })
 
   it('loads with a preset expiry (24 months)', async () => {
-    const wrapper: Wrapper<LimitedRestorationPanel> = createDefaultComponent(24)
-    await Vue.nextTick()
+    wrapper = createDefaultComponent(24)
+    await nextTick()
 
     expect(wrapper.vm.$data.radioValue).toEqual('24')
 
-    wrapper.destroy()
+    wrapper.unmount()
   })
 
   it('loads with a custom expiry (1 month)', async () => {
-    const wrapper: Wrapper<LimitedRestorationPanel> = createDefaultComponent(1)
-    await Vue.nextTick()
+    wrapper = createDefaultComponent(1)
+    await nextTick()
 
     expect(wrapper.vm.$data.radioValue).toEqual('customMonths')
     expect(wrapper.vm.$data.inputValue).toEqual('1')
 
-    wrapper.destroy()
+    wrapper.unmount()
   })
 
-  it('emits events when we select a preset expiry (24 months)', async () => {
-    const wrapper: Wrapper<LimitedRestorationPanel> = createDefaultComponent()
-    await Vue.nextTick()
+  it('emits correct events when preset expiry (24 months) radio is selected', async () => {
+    wrapper = createDefaultComponent()
+    await nextTick()
+
+    await wrapper.find('#radio-24').setChecked()
+    expect(wrapper.emitted()).toHaveProperty('valid')
+    expect(wrapper.emitted().valid).toBeTruthy()
+    expect(wrapper.emitted().months).toHaveLength(1)
+    expect(wrapper.vm.$data.radioValue).toEqual('24')
+
+    wrapper.unmount()
+  })
+
+  it('emits valid=false when input 25 months', async () => {
+    wrapper = createDefaultComponent()
+    await nextTick()
+
+    await wrapper.find('#radio-custom').setChecked()
+    await wrapper.find('#text-field-months').setValue('25')
+
+    expect(wrapper.vm.valid).toBeFalsy()
+
+    wrapper.unmount()
+  })
+
+  it.skip('emits events when we select a preset expiry (24 months)', async () => {
```

Comment:
What's the plan to fix this unit test (and the others below)?

Reply:
Frankly no plan as of now. I will create a tech debt ticket for this. May be sometime in near future will have to have heads down time to figure out all unit test fixes.

Reply:
Ok by me.

Reply:
Ketaki, please resolve this conversation after you've created the new ticket. Thx!

Reply:
Unit test fix ticket [19149](https://app.zenhub.com/workspaces/entities-team-space---new-65550db1fcfd2f002838a52a/issues/gh/bcgov/entity/19149)